libobs-d3d11/d3d11-subsystem.cpp

```diff
@@ -660,6 +687,19 @@ void gs_device::InitDevice(uint32_t adapterIdx)
 	blog(LOG_INFO, "Loading up D3D11 on adapter %s (%" PRIu32 ")",
 	     adapterNameUTF8.Get(), adapterIdx);
 
+	Name = adapterNameUTF8.Get();
+	DedicatedVideoMemory = desc.DedicatedVideoMemory;
+	SharedSystemMemory = desc.SharedSystemMemory;
+	VendorId = desc.VendorId;
+	DeviceId = desc.DeviceId;
```

Comment:
Everything except `VendorId` appears unused. These + `DriverVersion` and `CheckFastClear()` probably don't need to be part of the strcut and could all just be variables/static functions since it's only ever evaluated when initialising.

Reply:
As mentioned in another comment, I think they were holdovers from previous experimentation (the branch itself is from about 4 months ago).

As for why they are part of the struct, the values should never change, so I felt they logically should be associated with the device on init. My recollection was that we've had other cases where we wished we could gate something on a driver version, so I wanted to enable that to be easily reusable for other cases.

That said, I'm open to alternatives here.

Reply:
Until those are actually needed, they should probably just be left out to simplify the diff and save a couple bytes of memory.

Reply:
Can try to simplify the diff some time later.

Reply:
Here's what I'd cut it down to: https://github.com/derrod/obs-studio/commit/1cb357e1f64265ca38a694be2dd90de31e9dc060

Reply:
> Here's what I'd cut it down to: [derrod@1cb357e](https://github.com/derrod/obs-studio/commit/1cb357e1f64265ca38a694be2dd90de31e9dc060)

Used this implementation and squashed commits.